nautobot/dcim/models/device_components.py

```diff
@@ -519,6 +519,9 @@ def clean(self):
         if not self.mode and self.untagged_vlan is not None:
             raise ValidationError({"untagged_vlan": "Mode must be set when specifying untagged_vlan"})
 
+        if self.tagged_vlans.exists() and self.mode != InterfaceModeChoices.MODE_TAGGED:
```

Comment:
If validation logic for `untagged_vlan` and `tagged_vlans` is going to exist in `BaseInterface` then those fields should be moved to `BaseInterface` or the validation logic moved to the models where those fields exist? New issue perhaps?

Reply:
Would probably have to move to a mixin since they're related fields. It's probably not worth worrying about today.